File: apple/Sources/CCDAEngine/Models/Patient/CCDAAddress.swift
import Foundation

/// Postal address information from a C-CDA patient role or related party.
public struct CCDAAddress: Hashable, Sendable {
    /// C-CDA address use, such as home or work.
    public let use: CCDAAddressUse
    /// Street address lines in document order.
    public let streetLines: [String]
    /// City or locality.
    public let city: String?
    /// State, province, or region.
    public let state: String?
    /// Postal or ZIP code.
    public let postalCode: String?
    /// Country text or code.
    public let country: String?

    /// Creates a postal address value.
    public init(
        use: CCDAAddressUse = .unknown,
        streetLines: [String] = [],
        city: String? = nil,
        state: String? = nil,
        postalCode: String? = nil,
        country: String? = nil
    ) {
        self.use = use
        self.streetLines = streetLines
        self.city = city
        self.state = state
        self.postalCode = postalCode
        self.country = country
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Patient/CCDAAddress.kt
package com.ccdakit.engine

/**
 * Patient postal address parsed from `patientRole/addr`.
 *
 * @property use Typed address use code.
 * @property streetLines Street address lines.
 * @property city City name.
 * @property state State, province, or region.
 * @property postalCode Postal code.
 * @property country Country name or code.
 */
data class CCDAAddress(
    val use: CCDAAddressUse = CCDAAddressUse.Unknown,
    val streetLines: List<String> = emptyList(),
    val city: String? = null,
    val state: String? = null,
    val postalCode: String? = null,
    val country: String? = null,
)
